### PR TITLE
No more missing docs

### PR DESCRIPTION
**Pull request by killercup**
Sets `#![warn(missing_docs)]` and adds missing API docs. These were written without much knowledge of the actix-web internals! Please review carefully!

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -239,12 +244,14 @@ where
     }
 }
 
+/// Consume a future
 pub struct Drain<A> {
```
Comment (killercup) [inline src/context.rs:248]: I have no idea what this actually does

File: src/error.rs [hunk 2]
```diff
@@ -521,12 +521,16 @@ impl ResponseError for UriSegmentError {
 /// Errors which can occur when attempting to generate resource uri.
 #[derive(Fail, Debug, PartialEq)]
 pub enum UrlGenerationError {
+    /// Resource not found
     #[fail(display = "Resource not found")]
```
Comment (killercup) [inline src/error.rs:525]: Memo to self: Can't failure infer the display text from a doc comment?

**Comment by fafhrd91** [comment]
Oh! Thanks!

File: src/fs.rs [hunk 3]
```diff
@@ -422,15 +422,19 @@ type DirectoryRenderer<S> =
 /// A directory; responds with the generated directory listing.
 #[derive(Debug)]
 pub struct Directory {
+    /// Base directory
     pub base: PathBuf,
+    /// Path of subdirectory to generate listing for
```
Comment (killercup) [inline src/fs.rs:427]: Not a very good explanation

File: src/httpcodes.rs [hunk 4]
```diff
@@ -5,7 +5,7 @@ use httpresponse::{HttpResponse, HttpResponseBuilder};
 
 macro_rules! STATIC_RESP {
     ($name:ident, $status:expr) => {
-        #[allow(non_snake_case)]
+        #[allow(non_snake_case, missing_docs)]
```
Comment (killercup) [inline src/httpcodes.rs:8]: macros are evil but you could play with transforming the `$name:ident` to a `$name:expr` (via `$name:tt`) and putting it in a `#[doc = $name]` attr

**Review by fafhrd91 — APPROVED** [review]

**Review by killercup — COMMENTED** [review]
Most of these comments just give a little bit of context, but in the future could really come to shine if they also included examples and links to relevant guides or other APIs

**Comment by fafhrd91** [comment]
i think this is good

**Pull request merged**